CAIPs/caip-25.md

```diff
@@ -65,9 +65,13 @@ An example of a successful response should match the following format:
 {
     "id": 1,
     "jsonrpc": "2.0",
-    "result": {
-        "accounts": ["eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb"],
-    }
+    "result": [
+        {
+            "accounts": ["eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb"],
+            "methods": ["eth_sendTransaction", "eth_signTransaction", "eth_sign", "personal_sign"]
+            "events": ["accountsChanged", "chainChanged"]
+        }
+    ]
```

Comment:
I think this format makes sense! A couple of questions:
1. The wallet can return an arbitrary number of result objects in the array with any combination of accounts, methods, and events, right? I assume that's your intention here, I just want to confirm.
2. This response format only makes sense for methods and events that parameterize the account(s) and chain(s) in some way. However, in some cases only the chain is parameterized, and for other cases neither the account nor the chain are parameterized. Could we permit results of this form?

```
    "result": [
        // Account and chain parameterized
        {
            "accounts": ["eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb"],
            "methods": ["eth_sendTransaction", "eth_signTransaction", "eth_sign", "personal_sign"],
            "events": ["accountsChanged", "chainChanged"]
        },
        // Only the chain parameterized
        {
            "chains": ["eip155:10"],
            "methods": ["eth_blockNumber"],
            "events": ["chainChanged"]
        }
        // Neither account nor chain parameterized
        },
            "methods": ["wallet_installSnap", "wallet_getPermissions", "wallet_requestPermissions"],
            "events": ["wallet_locked", "wallet_unlocked"]
        }
    ]
```

3. The observation in (2) has implications for the `params` in the handshake request as well, specifically if the application requests methods for which the chain is not parameterized, like `wallet_getPermissions`. For that reason, could we permit omitting the `chains` property from the handshake param objects as well?